server/services/echo_service_test.go

```diff
@@ -38,23 +40,26 @@ func TestEcho_success(t *testing.T) {
 	server := NewEchoServer()
 	for _, val := range table {
 		in := &pb.EchoRequest{Response: &pb.EchoRequest_Content{Content: val}}
-		out, err := server.Echo(context.Background(), in)
+		mockStream := &mockUnaryStream{t: t}
+		out, err := server.Echo(appendTestOutgoingMetadata(context.Background(), &mockSTS{t: t, stream: mockStream}), in)
 		if err != nil {
 			t.Error(err)
 		}
 		if out.GetContent() != in.GetContent() {
 			t.Errorf("Echo(%s) returned %s", in.GetContent(), out.GetContent())
 		}
+		mockStream.verify(err != nil)
 	}
 	in := &pb.EchoRequest{
 		Response: &pb.EchoRequest_Error{
 			Error: &spb.Status{Code: int32(codes.OK)}}}
 
-	ctx := appendTestOutgoingMetadata(context.Background())
-	_, err := server.Echo(ctx, in)
+	mockStream := &mockUnaryStream{t: t}
+	_, err := server.Echo(appendTestOutgoingMetadata(context.Background(), &mockSTS{t: t, stream: mockStream}), in)
```

Comment:
Please put this context creation on its own line again :)

Reply:
you got it!